substrate/frame/treasury/src/lib.rs

```diff
@@ -71,6 +71,8 @@
 //! [`pallet::Config::PayoutPeriod`].
 
 #![cfg_attr(not(feature = "std"), no_std)]
+// not all specific usages can be marked as deprecated
+#![allow(deprecated)]
```

Comment:
try to get rid of this. something something like this helps
```rust 
#[allow(deprecated)]
{
  // deprecated invocation 
}
```

Reply:
Didn't work removing messages in all permutations of public types, I just removed the deprecation message in 4 public types here https://github.com/paritytech/polkadot-sdk/pull/6169/commits/4a35d3748fd46af1b2562fd9b787bb43c311940f